server/pipeline.go

```diff
@@ -7,57 +7,49 @@ import (
 	"github.com/pkg/errors"
 )
 
-// ScheduleNextTaskIfNeeded tries to schedule the next task if needed.
-// Returns nil if no task applicable can be scheduled.
-func (s *Server) ScheduleNextTaskIfNeeded(ctx context.Context, pipeline *api.Pipeline) (*api.Task, error) {
-	for _, stage := range pipeline.StageList {
-		for _, task := range stage.TaskList {
-			// Should short circuit upon reaching RUNNING or FAILED task.
-			if task.Status == api.TaskRunning || task.Status == api.TaskFailed {
-				return nil, nil
+// ScheduleActiveStageTask tries to schedule the tasks in the active stage.
+func (s *Server) ScheduleActiveStageTask(ctx context.Context, pipeline *api.Pipeline) error {
```

Comment:
```suggestion
// ScheduleActiveStageTasks tries to schedule the tasks in the active stage.
func (s *Server) ScheduleActiveStageTasks(ctx context.Context, pipeline *api.Pipeline) error {
```
Per the comment description and code, this should be in the plural form.

Reply:
Plurals are hard. I recall that we discourage plurals in our codebase. @tianzhou wdyt

Reply:
Discourage plural because it's easy to miss `s`, and some nouns share the same singular/plural form

In this case, I suggest ScheduleAllTasksInActiveStage

Reply:
Renamed to `ScheduleActiveStage`

Reply:
SG.